src/SFML/Audio/SoundFileReaderFlac.cpp

```diff
@@ -44,15 +44,17 @@ FLAC__StreamDecoderReadStatus streamRead(const FLAC__StreamDecoder*, FLAC__byte
 {
     auto* data = static_cast<sf::priv::SoundFileReaderFlac::ClientData*>(clientData);
 
-    const std::int64_t count = data->stream->read(buffer, static_cast<std::int64_t>(*bytes));
-    if (count > 0)
+    if (const auto count = data->stream->read(buffer, *bytes))
```

Comment:
```suggestion
    if (const auto count = data->stream->read(buffer, *bytes); count.has_value())
```

Nitpick, but this suggestion makes it obvious that `count` is an optional in this context

Reply:
I still'd like this one to be somehow addressed because in this context it's not really obvious if `count` is an optional or if it's an integer value and we're relying on implicit conversion from integer to bool.

If you don't like my current suggestion, consider also:

```cpp
if (const std::optional count = data->stream->read(buffer, *bytes))
if (const std::optional<std::size_t> count = data->stream->read(buffer, *bytes))
```